src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/OOPTagHelperResolver.cs

```diff
@@ -73,69 +72,50 @@ public override async Task<TagHelperResolutionResult> GetTagHelpersAsync(Project
                 TagHelperResolutionResult result = null;
                 if (factory != null)
                 {
-                    result = await ResolveTagHelpersOutOfProcessAsync(factory, workspaceProject, projectSnapshot).ConfigureAwait(false);
+                    result = await ResolveTagHelpersOutOfProcessAsync(factory, workspaceProject, projectSnapshot, cancellationToken).ConfigureAwait(false);
                 }
 
                 if (result == null)
                 {
                     // Was unable to get tag helpers OOP, fallback to default behavior.
-                    result = await ResolveTagHelpersInProcessAsync(workspaceProject, projectSnapshot).ConfigureAwait(false);
+                    result = await ResolveTagHelpersInProcessAsync(workspaceProject, projectSnapshot, cancellationToken).ConfigureAwait(false);
                 }
 
                 return result;
             }
             catch (Exception exception)
             {
-                throw new InvalidOperationException(
-                    Resources.FormatUnexpectedException(
-                        typeof(DefaultTagHelperResolver).FullName,
-                        nameof(GetTagHelpersAsync)),
-                    exception);
+                throw new InvalidOperationException($"An unexpected exception occurred when invoking '{typeof(DefaultTagHelperResolver).FullName}.{nameof(GetTagHelpersAsync)}' on the Razor language service.", exception);
             }
         }
 
-        protected virtual async Task<TagHelperResolutionResult> ResolveTagHelpersOutOfProcessAsync(IProjectEngineFactory factory, Project workspaceProject, ProjectSnapshot projectSnapshot)
+        protected virtual async Task<TagHelperResolutionResult> ResolveTagHelpersOutOfProcessAsync(IProjectEngineFactory factory, Project workspaceProject, ProjectSnapshot projectSnapshot, CancellationToken cancellationToken)
         {
             // We're being overly defensive here because the OOP host can return null for the client/session/operation
             // when it's disconnected (user stops the process).
             //
             // This will change in the future to an easier to consume API but for VS RTM this is what we have.
-            try
-            {
-#pragma warning disable CS0612 // Type or member is obsolete
-                var remoteClient = await RazorRemoteHostClient.CreateAsync(_workspace, CancellationToken.None);
-#pragma warning restore CS0612 // Type or member is obsolete
+            var remoteClient = await RazorRemoteHostClient.TryGetClientAsync(_workspace.Services, RazorServiceDescriptors.TagHelperProviderServiceDescriptors, RazorRemoteServiceCallbackDispatcherRegistry.Empty, CancellationToken.None);
```

Comment:
These are the new APIs we utilize. The rest is the "glue" to make sure these work under the covers 😄

Reply:
We pass CancellationToken.None here but the method takes a cancellationToken, why not use that?

Reply:
Totally legit issue. Fixed